pxr/imaging/hgiVulkan/texture.cpp

```diff
@@ -514,6 +522,73 @@ HgiVulkanTexture::CopyBufferToTexture(
         VK_PIPELINE_STAGE_ALL_GRAPHICS_BIT); // Consumer stage
 }
 
+void 
+HgiVulkanTexture::SubmitLayoutChange(HgiTextureUsage newLayout)
+{
+    VkImageLayout newVkLayout = 
+        HgiVulkanTexture::GetDefaultImageLayout(newLayout);
+
+    HgiVulkanCommandQueue* queue = _device->GetCommandQueue();
+    HgiVulkanCommandBuffer* cb = queue->AcquireResourceCommandBuffer();
+
+    VkAccessFlags srcAccessMask, dstAccessMask;
+
+    // The following cases are based on few initial assumptions to provide
+    // an infrastructure for access mask selection based on layouts.
+    // Feel free to update depending on need and use cases.
+    switch (GetImageLayout()) {
```

Comment:
This and the following switch statement need a default case, otherwise they fail to compile for me.